core/orchestrator.go

```diff
@@ -958,7 +958,24 @@ func (rtm *RemoteTranscoderManager) selectTranscoder(sessionId string, caps *Cap
 	return nil, ErrNoTranscodersAvailable
 }
 
-// compleStreamSessions end a stream session for a remote transcoder and decrements its laod
+// ends transcoding session and releases resources
+func (rtm *RemoteTranscoderManager) EndSession(sessionId string) {
+	// session may be already terminated by timeout
+	if sess, exists := rtm.streamSessions[sessionId]; exists {
+		// send NotifySegment with empty URL to signal session teardown
+		// xxx get valid auth token?
```

Comment:
Worth double checking, but...I think the AuthToken.Token field is only validated in [verifySegCreds()](https://github.com/livepeer/go-livepeer/blob/5d9bcaa325d8caa18151b42f50f4f78bf76834d7/server/segment_rpc.go#L399) which is only called by an O for segment requests from B. And in all other cases I think the AuthToken struct is just used to pass around the SessionId field.